core/tauri/src/lib.rs

```diff
@@ -387,14 +421,20 @@ impl Context {
 
   /// The assets to be served directly by Tauri.
   #[inline(always)]
-  pub fn assets(&self) -> &dyn Assets {
-    self.assets.as_ref()
+  pub fn assets(&self) -> &dyn Assets<R> {
+    self.assets.as_ref().unwrap().as_ref()
   }
 
   /// A mutable reference to the assets to be served directly by Tauri.
   #[inline(always)]
-  pub fn assets_mut(&mut self) -> &mut Box<dyn Assets> {
-    &mut self.assets
+  pub fn assets_mut(&mut self) -> &mut Box<dyn Assets<R>> {
+    self.assets.as_mut().unwrap()
```

Comment:
should we return Options here instead of unwrapping?

Reply:
it's an Option internally only for us to replace the value and return the previous one (so custom asset providers can fallback to the embedded assets if they desire)

Reply:
but they'll never be None

Reply:
but we provide `set_assets` method and so it could change by users and then panic on this unwrap later, better to be explicit here and also match other methods like `tray_icon` and `set_tray_icon`

Reply:
we'll remove this one, and even if the user use set_assets the value will never be None, we just want to be able to take() the previous value

Reply:
ah right, I was under the assumption that `set_assets` will take an `Option<Box<dyn Assets>>` and users will be allowed to set `None`.

Reply:
Should probably remove the `Option` internally and do the `.replace()` manually by calling `mem::replace`